src/Symfony/Component/HttpKernel/Tests/Controller/ContainerControllerResolverTest.php

```diff
@@ -106,6 +107,43 @@ public function testNonInstantiableController()
         $this->assertSame(array(NonInstantiableController::class, 'action'), $controller);
     }
 
+    public function testNonConstructController()
+    {
+        $container = $this->getMockBuilder(ContainerBuilder::class)->getMock();
+        $container->expects($this->at(0))
+            ->method('has')
+            ->with(ImpossibleConstructController::class)
+            ->will($this->returnValue(true))
+        ;
+
+        $container->expects($this->at(1))
+            ->method('has')
+            ->with(ImpossibleConstructController::class)
+            ->will($this->returnValue(false))
+        ;
+
+        $container->expects($this->atLeastOnce())
+            ->method('getRemovedIds')
+            ->with()
+            ->will($this->returnValue(array(ImpossibleConstructController::class)))
+        ;
+
+        if (method_exists($this, 'expectException')) {
```

Comment:
Could we use an annotation instead?

Reply:
Ill try but in other test we use this instead of annot, it's maybe because of lower version of php using lower version of phpunit.